secure_message/repository/retriever.py

```diff
@@ -16,16 +16,22 @@
 
 class Retriever:
     """Created when retrieving messages"""
+
     @staticmethod
     def retrieve_message_list(page, limit, user, ru_id=None, survey=None, cc=None, ce=None, label=None, descend=True):
         """returns list of messages from db"""
-        conditions = []
-        status_conditions = []
 
         if user.is_respondent:
-            status_conditions.append(Status.actor == str(user.user_uuid))
-        else:
-            status_conditions.append(Status.actor == constants.BRES_USER)
+            return Retriever._retrieve_message_list_respondent(page, limit, user=user, ru_id=ru_id, survey=survey,
```

Comment:
Not sure about the python way to do things, but when there ar eso many parameters is not better to encapsulate them all in a Object ?

Reply:
This could be made cleaner by refactoring the `is not None` below to just do dict key access on kwargs passed here. If a KeyError is raised, you know the key isn't passed and therefore can do the `else` condition from below.

Reply:
I agree with possibly moving the parameters to kwargs . However this how the existing code handles it. So whilst I agree its best to follow the boy scout principle , in this case I will raise a ticket to loop back depending on time.

Reply:
Why not use the named tuple created in the calling function instead? It seems to create the Named Tuple when the request arrives but then unpack it to seperate variables in order to call these function when actually everything you need (apart from the user) is in the tuple

Reply:
I am putting this on a tech debt ticket that I plan on coming back to asap